zerver/tests/test_send_email.py

```diff
@@ -70,39 +74,132 @@ def test_limited_to_length(self) -> None:
 
 
 class TestSendEmail(ZulipTestCase):
+    @override_settings(EMAIL_MAX_CONNECTION_LIFETIME_IN_MINUTES=None)
+    @time_machine.travel(timezone_now(), tick=False)
     def test_initialize_connection(self) -> None:
         # Test the new connection case
         with mock.patch.object(EmailBackend, "open", return_value=True):
             backend = initialize_connection(None)
             self.assertTrue(isinstance(backend, EmailBackend))
 
-        backend = mock.MagicMock(spec=SMTPBackend)
-        backend.connection = mock.MagicMock(spec=SMTP)
-
-        self.assertTrue(isinstance(backend, SMTPBackend))
-
-        # Test the old connection case when it is still open
-        backend.open.return_value = False
-        backend.connection.noop.return_value = [250]
-        initialize_connection(backend)
-        self.assertEqual(backend.open.call_count, 1)
-        self.assertEqual(backend.connection.noop.call_count, 1)
-
-        # Test the old connection case when it was closed by the server
-        backend.connection.noop.return_value = [404]
-        backend.close.return_value = False
-        initialize_connection(backend)
-        # 2 more calls to open, 1 more call to noop and 1 call to close
-        self.assertEqual(backend.open.call_count, 3)
-        self.assertEqual(backend.connection.noop.call_count, 2)
-        self.assertEqual(backend.close.call_count, 1)
-
-        # Test backoff procedure
-        backend.open.side_effect = OSError
-        with self.assertRaises(OSError):
+        backend = PersistentSMTPEmailBackend()
+        backend.open()
+        self.assertEqual(backend.opened_at, timezone_now())
+
+        with mock.patch.object(
+            backend, "_open", wraps=backend._open
+        ) as _open_call, mock.patch.object(backend, "close", wraps=backend.close) as close_call:
+            backend.connection = mock.MagicMock(spec=SMTP)
+            backend.connection.noop.return_value = [250]
+            initialize_connection(backend)
+
+            self.assertEqual(_open_call.call_count, 1)
+            self.assertEqual(backend.connection.noop.call_count, 1)
+
+            # Test the old connection case when it was closed by the server
+            backend.connection.noop.return_value = [404]
+            initialize_connection(backend)
+
+            # 2 more calls to open and 1 call to close
+            self.assertEqual(_open_call.call_count, 3)
+            self.assertEqual(close_call.call_count, 1)
+
+            # Test backoff procedure
+            _open_call.side_effect = OSError
+            with self.assertRaises(OSError):
+                initialize_connection(backend)
+            # 3 more calls to open as we try 3 times before giving up
+            self.assertEqual(_open_call.call_count, 6)
+
+    @time_machine.travel(timezone_now(), tick=False)
+    def test_max_connection_lifetime(self) -> None:
```

Comment:
Maybe break this up into a test for `EMAIL_MAX_CONNECTION_LIFETIME_IN_MINUTE=undef`, one for `EMAIL_MAX_CONNECTION_LIFETIME_IN_MINUTE=0`, and one for `EMAIL_MAX_CONNECTION_LIFETIME_IN_MINUTE=5` ?